### ISSUE TITLE
`Line::styled` not applying styles in ratatui 0.26.0

### ISSUE THREAD

**Issue by jwodder** [ratatui-org/ratatui#938]
Consider the following code:

```rust
use crossterm::{
    event::read,
    execute,
    terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
};
use ratatui::{prelude::*, widgets::*};
use std::io;

fn main() -> io::Result<()> {
    let mut terminal = init_terminal()?;
    terminal.draw(|frame| {
        frame.render_widget(
            Paragraph::new(Line::styled("Bold and brash", Style::new().bold())),
            frame.size(),
        );
    })?;
    let _ = read()?;
    reset_terminal()?;
    Ok(())
}

fn init_terminal() -> io::Result<Terminal<CrosstermBackend<io::Stdout>>> {
    let mut stream = io::stdout();
    execute!(stream, EnterAlternateScreen)?;
    enable_raw_mode()?;
    Terminal::new(CrosstermBackend::new(stream))
}

fn reset_terminal() -> io::Result<()> {
    disable_raw_mode()?;
    execute!(io::stdout(), LeaveAlternateScreen)?;
    Ok(())
}
```

with the following dependencies:

```toml
[dependencies]
crossterm = "0.27.0"
ratatui = "0.26.0"
```

When run on macOS Sonoma 14.2.1 in Terminal.app, whether directly in the Terminal (`TERM=xterm-256color`) or inside GNU Screen (`TERM=screen-256color`), no styling is applied:

<img width="617" alt="Screenshot 2024-02-08 at 09 18 01" src="https://github.com/ratatui-org/ratatui/assets/98207/8a92aa8f-a939-4126-8f06-44abd7c36a92">

However, if the ratatui version is set back to 0.25.0 and no other changes are made, then the styling is applied:

<img width="617" alt="Screenshot 2024-02-08 at 09 18 46" src="https://github.com/ratatui-org/ratatui/assets/98207/1587ed0c-0649-4732-a1d4-405be7ef47cd">

I do not see anything relevant listed in the v0.26.0 breaking changes, so I assume that this is some sort of unintended bug.

**Comment by jwodder** [comment]
Further experimentation reveals that:

- Rendering a styled `Line` directly without wrapping it in a `Paragraph` succeeds:

    ```rust
    frame.render_widget(
        // Shown as bold:
        Line::styled("Bold and brash", Style::new().bold()),
        frame.size(),
    );
    ```

- Rendering the `Paragraph::new(Line::from(vec!["Hello, ".into(), "world!".red()]))` example from the [`Paragraph::new()`](https://docs.rs/ratatui/latest/ratatui/widgets/struct.Paragraph.html#method.new) docs succeeds.

- Styling the string directly with `Paragraph::new(Line::from("Bold and brash".bold()))` succeeds.

**Comment by joshka** [comment]
We should have a 0.26.1 fix out in the next few days for this. Either https://github.com/ratatui-org/ratatui/pull/930 or https://github.com/ratatui-org/ratatui/pull/926 likely fixes the problem. Try pointing at main with a patch override https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section if you need it before then / to confirm the fix addresses your issue.

**Comment by joshka** [comment]
This is fixed in 0.26.1
<img width="463" alt="image" src="https://github.com/ratatui-org/ratatui/assets/381361/d02d4336-fe05-435f-a15d-fda5017cf487">
<img width="370" alt="image" src="https://github.com/ratatui-org/ratatui/assets/381361/e1bb1460-366c-4364-97a7-67a673f8f857">
